src/librustc/ty/util.rs

```diff
@@ -639,13 +646,24 @@ impl<'a, 'tcx> ty::TyS<'tcx> {
         !tcx.at(span).is_copy_raw(param_env.and(self))
     }
 
+    /// Checks whether values of this type `T` have a size known at
+    /// compile time (i.e., whether `T: Sized`). Lifetimes are ignored
+    /// for the purposes of this check, so it can be an
+    /// over-approximation in generic contexts, where one can have
+    /// strange rules like `<T as Foo<'static>>::Bar: Sized` that
+    /// actually carry lifetime requirements.
     pub fn is_sized(&'tcx self,
                     tcx_at: TyCtxtAt<'a, 'tcx, 'tcx>,
                     param_env: ty::ParamEnv<'tcx>)-> bool
     {
         tcx_at.is_sized_raw(param_env.and(self))
     }
 
+    /// Checks whether values of this type `T` implement the `Freeze`
+    /// trait -- frozen types are those that do not contain a
+    /// `UnsafeCell` anywhere.  This is a language concept used to
+    /// determine how to handle `static` values, the trait itself is
```

Comment:
Not just `static`s but also the semantics of `&T` (e.g. we tell LLVM `&T` is read-only iff `T: Freeze`). In general, whether shared values are truly immutable, or interior mutability is allowed.